LayoutTests/fast/forms/label/label-selection.html

```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
+<html>
+<script src="../../../resources/js-test.js"></script>
+
+<body>
+<label id="labelWithInput" for="inputText">Some, Text associated with input</label>
+<input id="inputText" type="text">
+</body>
+
+<script>
+description('Test the selection of label associated with input.');
+
+var labelElement = document.getElementById('labelWithInput');
+testWithDoubleClick(labelElement);
+
+shouldBeEqualToString('window.getSelection().toString()', 'Some');
+
+labelElement.style.display = 'none';
+
+function testWithDoubleClick(element)
+{
+    eventSender.mouseMoveTo(element.offsetLeft, element.offsetTop);
+    eventSender.mouseDown();
+    eventSender.mouseUp();
+    eventSender.mouseDown();
+    eventSender.mouseUp();
```

Comment:
Let's use `UIHelper.doubleClickAt`.

Reply:
@pxlcoder - It seems it will take me some time to figure out this test case (not good with JS). Should we land updated test in separate patch?

Reply:
You can add `<script src="../../../resources/ui-helper.js"></script>` on line 4.

And then replace this code with `UIHelper.doubleClickAt(element.offsetLeft, element.offsetTop)`.

Reply:
Seems like this is now good to go and thanks again for your help as always. Really appreciate it.. Now let's make next Safari release awesome..